test/ConstexprUtilsTest.cpp

```diff
@@ -237,14 +252,26 @@ TEST(ConstexprUtils, ForEachTypeInParameterPack) {
   ASSERT_TRUE(typeToStringVector.empty());
 }
 
+struct TestConstExprForEachNormalCallForEachTypeInVariantTemplateType {
+  template <typename... Ts>
+  void operator()(const auto& func) const {
+    forEachTypeInTemplateType<std::variant<Ts...>>(func);
+  }
+};
+
+struct TestConstExprForEachNormalCallForEachTypeInTupleTemplateType {
+  template <typename... Ts>
+  void operator()(const auto& func) const {
+    forEachTypeInTemplateType<std::tuple<Ts...>>(func);
+  }
+};
+
 TEST(ConstexprUtils, forEachTypeInTemplateType) {
   // Normal call with `std::variant`.
-  testConstExprForEachNormalCall([]<typename... Ts>(const auto& func) {
-    forEachTypeInTemplateType<std::variant<Ts...>>(func);
-  });
+  testConstExprForEachNormalCall(
+      TestConstExprForEachNormalCallForEachTypeInVariantTemplateType{});
 
   // Normal call with `std::tuple`.
-  testConstExprForEachNormalCall([]<typename... Ts>(const auto& func) {
-    forEachTypeInTemplateType<std::tuple<Ts...>>(func);
-  });
+  testConstExprForEachNormalCall(
+      TestConstExprForEachNormalCallForEachTypeInTupleTemplateType{});
```

Comment:
I think it is worth rewriting the `ofrEachTypeInTemplateType` to get the template argument via `type_identity`. Because otherwise we always inhibit the nicer syntax in C++ 17 mode.

Reply:
Actually only partially addressed, since these tests are meant to test the functions in `ConstexprUtils.h`.
BTW I renamed the structures to be less verbose and added a couple of tests for the new functions.

Reply:
Yes, I have seen it and it looks fine as is, thank you very much.